python/lsst/daf/butler/dimensions/_config.py

```diff
@@ -156,6 +156,32 @@ def _updateWithOtherConfigFile(self, file: Config | str | ResourcePath | Mapping
         externalConfig = type(self)(file, validate=False)
         self.update(externalConfig)
 
+    def to_simple(self) -> SerializedDimensionConfig:
+        """Convert this configuration to a serializable Pydantic model.
+
+        Returns
+        -------
+        model : `SerializedUniverseConfig`
+            Serializable Pydantic version of this configuration.
+        """
+        return SerializedDimensionConfig.model_validate(self.toDict())
+
+    @staticmethod
+    def from_simple(simple: SerializedDimensionConfig) -> DimensionConfig:
```

Comment:
I think in other places we prefer to use `@classmethod` for this so you can do `cls(...)` to support potential subclassing. Not sure it matters here though.

Reply:
Generally speaking I'm not a big fan of that pattern, since it assumes subclasses have the same constructor signature as the base class.

And yeah specifically here no subclasses exist or should exist (and if they did, they would have a different serialized form with their own data so the function signatures wouldn't match up.)